src/v/config/constraint_method_generators.cc

```diff
@@ -0,0 +1,561 @@
+    // signed/unsigned int comparison which is avoidable.
+    try {
+        return std::get<range_values<RangeT>>(args.flags);
+    } catch (const std::bad_variant_access& ex) {
+        // This should not happen since the variant is hardcoded in
+        // constraint_types.cc, but just-in-case, let's catch and log.
+        logger.error(
+          "Constraints failure[min/max not found]: topic property {}.{}",
+          topic(),
+          args.name);
+        return range_values<RangeT>{};
+    }
+}
+
+template<typename RangeT, typename T>
+bool validate_range(
+  const T& topic_val, const model::topic& topic, const constraint_args& args) {
+    range_values<RangeT> range = get_range<RangeT>(args, topic);
+    if (is_invalid_range<RangeT, T>(range, topic_val)) {
+        logger.error(
+          "Constraints failure[value out-of-range]: topic property "
+          "{}.{}, value {}",
+          topic(),
+          args.name,
+          topic_val);
+        return false;
+    }
+
+    return true;
+}
+
+template<typename RangeT, typename T>
+std::optional<T> get_min_max(const range_values<RangeT>& range, T& topic_val) {
+    if (less_than_min<RangeT, T>(range.min, topic_val)) {
+        return std::make_optional(static_cast<T>(*range.min));
+    }
+
+    if (greater_than_max<RangeT, T>(range.max, topic_val)) {
+        return std::make_optional(static_cast<T>(*range.max));
+    }
+
+    return std::nullopt;
+}
+
+template<typename RangeT, typename T>
+void maybe_clamp_range(
+  T& topic_val, const model::topic& topic, const constraint_args& args) {
+    range_values<RangeT> range = get_range<RangeT>(args, topic);
+    auto min_max = get_min_max<RangeT, T>(range, topic_val);
+    if (min_max) {
+        logger.warn(
+          "Overwriting topic property {}.{} to the constraint min/max",
+          topic(),
+          args.name);
+        topic_val = *min_max;
+    }
+}
+
+template<typename T>
+bool match_cluster_property(
+  const T& topic_val,
+  const T& cluster_val,
+  const std::variant<
+    range_values<int64_t>,
+    range_values<uint64_t>,
+    constraint_enabled_t>& flags) {
+    return ss::visit(
+      flags,
+      [](const range_values<int64_t>) { return false; },
+      [](const range_values<uint64_t>) { return false; },
+      [&topic_val, &cluster_val](const constraint_enabled_t enabled) {
+          return enabled == constraint_enabled_t::yes
+                 && topic_val == cluster_val;
+      });
+}
+
+template<typename T>
+bool validate_property(
+  const T& topic_val,
+  const T& cluster_val,
+  const model::topic& topic,
+  const constraint_args& args) {
+    if (!match_cluster_property(topic_val, cluster_val, args.flags)) {
+        logger.error(
+          "Constraints failure[does not match the cluster default {}]: "
+          "topic property {}.{}, value {}",
+          cluster_val,
+          topic(),
+          args.name,
+          topic_val);
+        return false;
+    }
```

Comment:
I think this errors if `constraint_enabled_t` value is false (no), correct?  Is this the correct behavior?  If it isn't enabled wouldn't this succeed?

Reply:
Addressed the bug in the latest push.